.github/prompts/dotnet/async.prompt.md

```diff
@@ -0,0 +1,45 @@
+﻿This cheat sheet should serve as a quick reminder of best practices when writing asynchronous code. Following these guidelines will help you avoid common pitfalls such as unobserved exceptions, deadlocks, and unexpected UI blocking.
+
+---
+
+### ✅ DOs
+
+- **Always Await Your Tasks:**  
+  Always use the `await` keyword on async operations to ensure exceptions are captured and to avoid blocking the calling thread.  
+
+- **Use Async Task/Task<T> Methods:**  
+  Prefer returning `Task` or `Task<T>` over `async void` so that exceptions can be observed, and methods are easily composable and testable.  
+
+- **Name Methods with the "Async" Suffix:**  
+  Clearly differentiate asynchronous methods (e.g., `GetDataAsync()`) from synchronous ones.  
```

Comment:
we should add that this is needed if there's a synchronous implementation of that method.